### ISSUE TITLE
@RunAs annotation and Authz topic in OL docs

### ISSUE THREAD

**Issue by lauracowen** [OpenLiberty/docs#3406]
@dmuelle I think the OL Authz topic does a nice job of showing how to use the @RunAs annotation. I was wondering if any of the [conceptual info in the KC Authn topic](https://www.ibm.com/support/knowledgecenter/SSAW57_liberty/com.ibm.websphere.wlp.nd.multiplatform.doc/ae/cwlp_authentication.html#cwlp_authentication__run_as) would be useful as a sentence or two intro to that section? The KC info gives a little more context of why you'd use it (but I haven't read the OL Authz topic in detail lately so you may have already covered that).

**Comment by dmuelle** [comment]
Changes were approved by @arkarkala on Slack

pasting review relevant bits of convo here:


Ajay Reddy:

also here - <run-as userid=“Wanda” password=“{xor}Lz4sLCgwLTs=“/>. ---> password is optional and folks do not prefer that. just id is fine

but good to show an example w/o password

David Mueller  1:35 PM
point taken re "caller identity". One question I had, although you can use RunAs to designate a particular caller, the javadoc for theRunAs annotation defines it only in the context of running under a different security role.
"This allows developers to execute an application under a particular role"
Doesn't really discuss the second part, where you can configure that role in the server config to RunAs a particular caller.
Are there cases where someone might designate calls from a Servlet to @RunAs as a particular role, without having that role run as a specific caller? And if so, is "caller identity" the right term in those cases? That's why I had both security role and caller identity in the doc. Wondering if there might be cases where it's just the role that gets delegated, and not a specific ID.

Ajay:
correct. so the spec says -runAsRole - but role by itself does not mean anything until a user is associated with that role which is the caller identity
it leaves to the implementation to do the role to caller/user mapping
without a caller/identity, we would not know which user to use - role by itself does not make sense - so we always have a caller



Ajay:
some minor things - calls to downstream resources, such as other servlets.  ----> can say --> calls to downstream resources, such as other servlets and EJBs
I think everything else is fine. (edited)

**Comment by ManasiGandhi** [comment]
@dmuelle The Authorization topic looks good. A few edit suggestions after content review:

**Define roles in your application code**

- "If the role names match the group names exactly, the server maps the group to the role with a matching name by default. If you use this default mapping, you do not need to specify an application-bnd element in the server.xml file." Is this info documented elsewhere for reference?
- Alternatively, you can authorize all users by setting the @PermitAll annotation, which specifies that anyone can access a resource, even users who are not authenticated." Can you change "alternatively" to something else. @RolesAllowed  and @PermitAll annotations are two choices for the developers but are they alternatives of each other?

**Authorize requests between web services**

- Change "JSON Web Tokens (JWT)", to "a JSON Web Token (JWT)"
- Can you add some explanation about the dropins folder as there is not previous reference ?

**Comment by dmuelle** [comment]
Thanks for reviewing @ManasiGandhi , I made the following changes per your review:

- [ ] "If the role names match the group names exactly, the server maps the group to the role with a matching name by default. If you use this default mapping, you do not need to specify an application-bnd element in the server.xml file." Is this info documented elsewhere for reference? **No, this is the primary discussion of role biding in the OL docs currently.**
- [x] Alternatively, you can authorize all users by setting the @PermitAll annotation---> You can also authorize all users by setting the @PermitAll annotation
- [x] Change "JSON Web Tokens (JWT)", to "a JSON Web Token (JWT)"
- [x] Can you add some explanation about the dropins folder as there is not previous reference ? **Added explanation of the dropins folder and moved this statement up to the "Define roles in your application code" section. The info isn't specific to RunAs, so I think it's better mentioned earlier in the doc, where security roles are initially discussed.**

**Comment by Rwalls1** [comment]
# Peer review
The topic looks good, I just have a few comments that are mostly related to previous sections.

## Intro
- The phrase “Authorization is the process by which an application” seems a bit wordy, maybe change to something like, “Authorization of an application” to be more concise.

## Define roles in your application code
- I think the section title,”Define roles in your application code” could be shortened to “Define roles in your application” as I think “code” doesn’t doesn’t add any value within the context of a title
- For the phrase, "setting the `@RolesAllowed` annotation in the source code”, I think you could probably be a bit more specific than just stating “source code”. Is there a specific file within the application that the annotation is added to? Is the annotation set in the associated API? I think specifying the location could be helpful if possible.
- I don’t think the phrase “Specify the `security-role` element” aligns with the tone of the rest of the topic. I think rather than using a declarative statement, you should rephrase as something like, “To map the roles, you can specify the `security role` element”.
- I think the sentence “The `name` attribute of the `security-role element `corresponds to a role that is specified in the application” is missing an important detail regarding the behavior of the application. I think it is important to mention that rather than the role just being specified in the application, an annotation is specified that tells the application that the role is allowed. So I think you should change that to something like, “The `name` attributes of the `security-role` elements correspond to the roles that are specified in the application as permissible.”
- I you should change the phrase, “you can specify `**` to “you can specify the `**` value.
- For the phrase, “setting the `@PermitAll` annotation, I think you should also specify where the annotation is set
- I think the phrase “drop an application” could be a bit confusing. I think you should change it to something like “put an application” or “move an application”.
- The phrase, “such as security role mapping” doesn’t make sense to me in the context that is used. Can you review and clarify? Maybe there’s a typo?

## Configure servlet security roles
- For the phrase, `configuring the `@ServletSecurity` annotation, I think you should indicate where the annotation is configured.
- Typo: Change "roles with rolesAllowed attribute” to “roles with the `rolesAllowed` attribute.

## Configure calls to run under a specific caller identity
- I think you should change the phrase, “require a particular caller identity, Wanda” to "require a particular caller identity that is specified as “Wanda” for clarity purposes
- I think you should change the phrase “use Wanda’s caller identity” to “that caller identity” to avoid referring to Wanda as an actual person rather than just an name value that is specified in the example.
- For the phrase, “Wanda and Trevor are both in the `admin` security role”, I think you should change it to something like “the `“Wanda”` and `“Trevor”` name values are both specified as users that have the `admin` security role” to increase clarity.

**Comment by dmuelle** [comment]
Thanks for reviewing @Rwalls1 , I updated the doc per your review

## Intro
- [x] The phrase “Authorization is the process by which an application” seems a bit wordy, maybe change to something like, “Authorization of an application” to be more concise.

## Define roles in your application code
- [x] I think the section title,”Define roles in your application code” could be shortened to “Define roles in your application” 
- [x]  For the phrase, "setting the `@RolesAllowed` annotation in the source code”, I think you could probably be a bit more specific than just stating “source code”. Is there a specific file within the application that the annotation is added to? Is the annotation set in the associated API? **The role can be set on a class or method in the app code**
- [x] I don’t think the phrase “Specify the `security-role` element” aligns with the tone of the rest of the topic. I think rather than using a declarative statement, you should rephrase as something like, “To map the roles, you can specify the `security role` element”.
- [x] I think the sentence “The `name` attribute of the `security-role element `corresponds to a role that is specified in the application” is missing an important detail regarding the behavior of the application. I think it is important to mention that rather than the role just being specified in the application, an annotation is specified that tells the application that the role is allowed. So I think you should change that to something like, “The `name` attributes of the `security-role` elements correspond to the roles that are specified in the application as permissible.”
- [x] I you should change the phrase, “you can specify `**` to “you can specify the `**` value.
- For the phrase, “setting the `@PermitAll` annotation, I think you should also specify where the annotation is set
- [x] I think the phrase “drop an application” could be a bit confusing. I think you should change it to something like “put an application” or “move an application”.
- [x] The phrase, “such as security role mapping” doesn’t make sense to me in the context that is used. Can you review and clarify? Maybe there’s a typo?

## Configure servlet security roles
- [x] For the phrase, `configuring the `@ServletSecurity` annotation, I think you should indicate where the annotation is configured.
- [x] Typo: Change "roles with rolesAllowed attribute” to “roles with the `rolesAllowed` attribute.

## Configure calls to run under a specific caller identity
- [x] I think you should change the phrase, “require a particular caller identity, Wanda” to "require a particular caller identity that is specified as “Wanda” for clarity purposes
- [x] I think you should change the phrase “use Wanda’s caller identity” to “that caller identity” to avoid referring to Wanda as an actual person rather than just an name value that is specified in the example.
-  [ ] For the phrase, “Wanda and Trevor are both in the `admin` security role”, I think you should change it to something like “the `“Wanda”` and `“Trevor”` name values are both specified as users that have the `admin` security role” to increase clarity.

**Comment by Rwalls1** [comment]
@dmuelle Thanks, your changes look good.

**Comment by dmuelle** [comment]
content is on vNext, will publish with 21.0.0.3, Updated KC tracking spreadsheet
